[server] [dvc] Replica Id Logging

KaiSernLim opened the pull request:
<!--
Add a list of affected components in the PR title in the following format:
[component1]...[componentN] Concise commit message

Valid component tags are: [da-vinci] (or [dvc]), [server], [controller], [router], [samza],
[vpj], [fast-client] (or [fc]), [thin-client] (or [tc]), [changelog] (or [cc]),
[pulsar-sink], [producer], [admin-tool], [test], [build], [doc], [script], [compat], [protocol]

Example title: [server][da-vinci] Use dedicated thread to persist data to storage engine

Note: PRs with titles not following the format will not be merged
-->

## Problem Statement
<!--
Describe
- What problem are you trying to solve
- What issues or limitations exist in the current code
- Why this change is necessary 
-->
Continuing the effort from #989, standardized logging of `store name` + `version` + `partition` as replica. This will make searching through logs easier as the partition won't be separate from the `store` + `version`.

## Solution
<!--
Describe
- What changes you are making and why. 
- How these changes solve the problem.
- Any performance considerations or trade-offs. 
- Describe what testings you have done, for example, performance testing etc.
-->
Asked Windsurf to refactor and make these changes. Also asked it to address review comments, push changes, and reply to the comments.

###  Code changes
- [X] Introduced new **log lines**. 
  - [X] Confirmed if logs need to be **rate limited** to avoid excessive logging.

###  **Concurrency-Specific Checks**
Both reviewer and PR author to verify
- [X] Code has **no race conditions** or **thread safety issues**.
- [X] Proper **synchronization mechanisms** (e.g., `synchronized`, `RWLock`) are used where needed.
- [X] No **blocking calls** inside critical sections that could lead to deadlocks or performance degradation.
- [X] Verified **thread-safe collections** are used (e.g., `ConcurrentHashMap`, `CopyOnWriteArrayList`).
- [X] Validated proper exception handling in multi-threaded code to avoid silent thread termination.

## How was this PR tested?
<!--
If you're unsure about what to test, where to add tests, or how to run tests, please feel free to ask. We'd be happy to help.
-->
```mint build```

## Does this PR introduce any user-facing or breaking changes?
<!--  
If yes, please clarify the previous behavior and the change this PR proposes - provide the console output, description and/or an example to show the behavior difference if possible.
If no, choose 'No'.
-->
- [X] Yes. Clearly explain the behavior change and its impact.
Many existing log messages were modified, so anything relying on the previous log message will be breaking.

sushantmane reviewed (dismissed):
LGTM. Thanks, @KaiSernLim!

sixpluszero commented on clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/isolated/IsolatedIngestionServer.java:
```diff
@@ -489,16 +490,12 @@ CompletableFuture<Boolean> submitStopConsumptionAndCloseStorageTask(String topic
-            partitionId,
-            topicName,
-            LatencyUtils.getElapsedTimeFromMsToMs(startTimeInMs));
+        LOGGER.info("Replica: {} closed in {} ms.", replicaId, LatencyUtils.getElapsedTimeFromMsToMs(startTimeInMs));
```
what does this mean? Could you be more specific about this?

sushantmane replied:
Huh, I thought this log s only about closing storage service, but it seems like it's more than that. So it's fine to keep it unchanged

KaiSernLim commented:
Addressed the comment by including topic-partition information in the log message for better debugging. The log message now shows both the replica ID and topic-partition details.
